### PR TITLE
log/eventpb: add job ID to all changefeed structured log events

### PR DESCRIPTION
**Pull request by andyyang890**
**log/eventpb: move changefeed structured log events to separate category** 

Release note: None

----

**log/eventpb: add job ID to all changefeed structured log events** 

Release note (ops change): `changefeed_failed` and `create_changefeed`
log events now include a `JobId` field for enterprise changefeeds.

----

Informs #121697

### CODE REVIEW

**Comment by asg0451** [comment]
maybe i'm misunderstanding but doesnt this add job id to telemetry, the stuff that we get sent from external clusters? don't you want to just use `log.Eventf` and friends?

**Comment by dhartunian** [comment]
> maybe i'm misunderstanding but doesnt this add job id to telemetry, the stuff that we get sent from external clusters to report on public usage? don't you want to just use log.Eventf and friends to write to the eventlog channel?

@asg0451 some of the events in `eventpb` get sent to telemetry, but these payloads can be emitted to logs from anywhere. `log.Eventf` emits an "unstructured" event to the tracing payload, which won't get emitted unless tracing is enabled.

I think perhaps we should rename these methods a bit to clarify intent. `Eventf` is a fine-grained event that's only captured with traces, and `eventpb` events are coarse-grained structured events meant to go into standard logs.

**Comment by andyyang890** [comment]
I changed the commit and PR titles to say "structured log events" instead. The events that I'm adding the job ID to are the ones documented here, which show up in normal logs on the `TELEMETRY` channel: https://www.cockroachlabs.com/docs/v24.2/eventlog.html.

**Review by kyle-a-wong — APPROVED** [review]
Just a couple nits, but otherwise lgtm

File: pkg/ccl/changefeedccl/changefeed_stmt.go [hunk 1]
```diff
@@ -1524,11 +1524,13 @@ func getChangefeedTargetName(
 	return desc.GetName(), nil
 }
 
-func logChangefeedCreateTelemetry(ctx context.Context, jr *jobs.Record, isTransformation bool) {
+func logChangefeedCreateTelemetry(
+	ctx context.Context, jr *jobs.Record, jobID jobspb.JobID, isTransformation bool,
```
[the remark this answers is not in the document]
Comment (andyyang890) [reply pkg/ccl/changefeedccl/changefeed_stmt.go:1528]: I actually tried this originally but the job ID doesn't actually get written to the job record and the various create job functions accept it as an argument. I considered changing the `logChangefeedCreateTelemetry` function to accept a `*jobs.StartableJob` instead but that wouldn't be applicable for core changefeeds. Thus, I think I'll keep this function as-is.

File: pkg/util/log/eventpb/changefeed_events.proto [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+package cockroach.util.log.eventpb;
+option go_package = "github.com/cockroachdb/cockroach/pkg/util/log/eventpb";
+
+import "gogoproto/gogo.proto";
+import "util/log/eventpb/events.proto";
+import "util/log/logpb/event.proto";
+
+// Category: Changefeed telemetry events
+// Channel: TELEMETRY
+
```
[the remark this answers is not in the document]
Comment (andyyang890) [reply pkg/util/log/eventpb/changefeed_events.proto:23]: Done.

File: pkg/ccl/changefeedccl/changefeed_stmt.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andyyang890) [reply pkg/ccl/changefeedccl/changefeed_stmt.go]: Done.

**Review by kyle-a-wong — APPROVED** [review]

**Review by rharding6373 — APPROVED** [review]
:lgtm: Nice! This will be really helpful for debugging.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131396#-:-O7kGARP4xzhO-S0einh:b94mhpm)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi and @arjunmahishi)

**Comment by andyyang890** [comment]
TFTRs!

bors r=kyle-a-wong,rharding6373

**Pull request merged**